packages/editor-ui/src/views/NodeView.vue

```diff
@@ -1253,6 +1253,31 @@ export default mixins(
 							this.nodeSelectedByName(node.name);
 						});
 					});
+
+					const tagsEnabled = this.$store.getters['settings/areTagsEnabled'];
+					if (Array.isArray(workflowData.tags) && tagsEnabled) {
+						const allTags: ITag[] = await this.$store.dispatch('tags/fetchAll');
+						const tagNames = new Set(allTags.map((tag) => tag.name));
+
+						const workflowTags = workflowData.tags as ITag[];
+						const notFound = workflowTags.filter((tag) => !tagNames.has(tag.name));
+						for (const tag of notFound) {
+							allTags.push(await this.$store.dispatch('tags/create', tag.name));
+						}
+
+						const tagIds = workflowTags.map((imported) => {
+							const tag = allTags.find(tag => tag.name === imported.name);
+							if (tag) {
+								return tag.id;
+							}
+
+							return undefined;
+						})
+							.filter((id) => !!id);
```

Comment:
You can combine `map` and `filter` into a single `reduce`. This also allows you to remove `|| []` at L1278.